src/plugins/camera/camera_impl.cpp

```diff
@@ -726,6 +740,15 @@ void CameraImpl::process_camera_capture_status(const mavlink_message_t& message)
             (camera_capture_status.image_status == 2 || camera_capture_status.image_status == 3);
         _status.received_camera_capture_status = true;
         _status.data.recording_time_s = float(camera_capture_status.recording_time_ms) / 1e3f;
+        _status.image_count = camera_capture_status.image_count;
+
+        if (_status.image_count_at_connection == -1) {
+            _status.image_count_at_connection = camera_capture_status.image_count;
+        }
+
+        if (_status.last_advertised_image_index == -1) {
+            _status.last_advertised_image_index = camera_capture_status.image_count - 1;
```

Comment:
Why -1?

Reply:
Because an image index is the image count - 1 (if there are 3 images, the last image is number 2). At least I think it works :sweat_smile:

Reply:
Ok, got it.